src/MessagePack/Resolvers/DynamicGenericResolver.cs

```diff
@@ -74,16 +71,24 @@ internal static class DynamicGenericResolverGetFormatterHelper
               { typeof(IReadOnlyList<>), typeof(InterfaceReadOnlyListFormatter<>) },
               { typeof(IReadOnlyCollection<>), typeof(InterfaceReadOnlyCollectionFormatter<>) },
               { typeof(ISet<>), typeof(InterfaceSetFormatter<>) },
-#if NET5_0_OR_GREATER
-              { typeof(IReadOnlySet<>), typeof(InterfaceReadOnlySetFormatter<>) },
-#endif
+
               { typeof(System.Collections.Concurrent.ConcurrentBag<>), typeof(ConcurrentBagFormatter<>) },
               { typeof(System.Collections.Concurrent.ConcurrentQueue<>), typeof(ConcurrentQueueFormatter<>) },
               { typeof(System.Collections.Concurrent.ConcurrentStack<>), typeof(ConcurrentStackFormatter<>) },
               { typeof(ReadOnlyDictionary<,>), typeof(ReadOnlyDictionaryFormatter<,>) },
               { typeof(IReadOnlyDictionary<,>), typeof(InterfaceReadOnlyDictionaryFormatter<,>) },
               { typeof(System.Collections.Concurrent.ConcurrentDictionary<,>), typeof(ConcurrentDictionaryFormatter<,>) },
               { typeof(Lazy<>), typeof(LazyFormatter<>) },
+#if NET5_0_OR_GREATER
+              { typeof(IReadOnlySet<>), typeof(InterfaceReadOnlySetFormatter<>) },
+#endif
+#if NET6_0_OR_GREATER
+              { typeof(PriorityQueue<,>), typeof(PriorityQueueFormatter<,>) },
+#endif
```

Comment:
IMO you can just group both of these under `#if NET` because 5 and 6 are not supported any more, so anything in .NET 8 can be simply `if NET`

Reply:
That's true.
However, it's not a bad thing to have a clear time for adding it.
Also, I don't think they'll integrate it every time they turn off support, so I'll leave it as it is.